src/Symfony/Component/Security/Http/Firewall/ContextListener.php

```diff
@@ -76,7 +76,10 @@ public function __construct(TokenStorageInterface $tokenStorage, iterable $userP
      */
     public function setLogoutOnUserChange($logoutOnUserChange)
     {
-        $this->logoutOnUserChange = (bool) $logoutOnUserChange;
+        // to be deprecated in 4.1
+        if (true !== $logoutOnUserChange) {
+            throw new \RuntimeException(sprintf('The %s() should not be used with a value other than true.', __METHOD__));
```

Comment:
Same should be done in the bundle configuration

Reply:
In fact, I am not sure anymore if we should really throw the exception here. I mean this is now a code path that changes its behaviour compared to Symfony 3.4 without triggering a deprecation before.

Reply:
In theory this would shield it when people set it to false and didn't adhere to the deprecation from 3.4. In theory nothing breaks though, except that a useless method call is executed.

Could instead make it trigger just a deprecation in 4.1